include/mata/nfa/nfa.hh

```diff
@@ -161,12 +161,31 @@ public:
      * @brief Get the useful states using a modified Tarjan's algorithm. A state
      * is useful if it is reachable from an initial state and can reach a final state.
      *
-     * @param state_at_first_useful_state Whether only the first found useful state is set to true. Useful when
-     *  checking whether the set of useful states is empty.
-     *
      * @return BoolVector Bool vector whose ith value is true iff the state i is useful.
      */
-    BoolVector get_useful_states(const bool stop_at_first_useful_state = false) const;
+    BoolVector get_useful_states() const;
+
+    /**
+     * @brief Structure for utilizing Tarjan's SCC discover.
+     * 
+     */
+    struct SCCCrawlExe {
```

Comment:
I think a more thorough description would be beneficial. What does it do? What does the name mean? Could we think of a more descriptive name?

Reply:
I updated the comment a bit and renamed stuff.